server/src/com/cloud/user/AccountManagerImpl.java

```diff
@@ -2632,4 +2651,27 @@ public void checkAccess(Account account, DiskOffering dof)
         assert false : "How can all of the security checkers pass on checking this caller?";
         throw new PermissionDeniedException("There's no way to confirm " + account + " has access to " + dof);
     }
+
+    @Override
+    public void checkAccess(User user, ControlledEntity entity)
+        throws PermissionDeniedException {
+        for(SecurityChecker checker : _securityCheckers){
+            if(checker.checkAccess(user,entity)){
+                if(s_logger.isDebugEnabled()){
+                    s_logger.debug("Access granted to " + user + "to " + entity + "by " + checker.getName());
+                }
+                return;
```

Comment:
I am not sure about it. But if you look at all the checkAccess methods in this file has the same pattern.